appinventor/appengine/src/com/google/appinventor/server/DownloadServlet.java

```diff
@@ -268,6 +289,15 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOExc
       } else {
         throw new IllegalArgumentException("Unknown download kind: " + downloadKind);
       }
+
+      if (!isInline) {
+        final String downloadObjectUrl = shouldUseRemoteStorageDownload(downloadKind, userId, downloadableFile);
```

Comment:
I couldn't really find where the inline mode is used, but I think it's safe to assume that if something should render "inline", it's better to not use foreign origins or anything similar...